exes/button/src/main.cpp

```diff
@@ -0,0 +1,153 @@
+#include <cstdlib>
+#include <iostream>
+
+#include <boost/asio.hpp>
+#include <boost/program_options.hpp>
+#include <boost/sml.hpp>
+
+#include <tfc/ipc.hpp>
+#include <tfc/logger.hpp>
+#include <tfc/progbase.hpp>
+
+namespace asio = boost::asio;
+
+namespace sml = boost::sml;
+using sml::operator""_s;
+using sml::operator""_e;
+
+using std::chrono::milliseconds;
+using std::chrono_literals::operator""ms;
+using std::chrono::system_clock;
+
+// Events
+struct high {};
+struct low {};
+struct timeout {};
+
+struct my_deps {
+  tfc::ipc::bool_signal& long_signal;
+  tfc::ipc::bool_signal& double_signal;
+  tfc::ipc::bool_signal& touch_signal;
+  asio::io_context& ctx;
+  tfc::logger::logger& logger;
+  std::chrono::time_point<system_clock, std::chrono::nanoseconds> initial_time;
+};
+
+// guards
```

Comment:
lambdas can be constexpr

Reply:
I will make them constexpr. Is there anything specific you hope to gain in this case?

Reply:
not being globally constructed/destructed, instead being part of the compiled artifact(a symbol). I thought this hsould provoke an clang warning